components/butterfly/src/rumor/mod.rs

```diff
@@ -431,10 +431,30 @@ where
 
     pub fn with_rumor<F>(&self, key: &str, member_id: &str, mut with_closure: F)
     where
-        F: FnMut(Option<&T>),
+        F: FnMut(&T),
     {
         let list = self.list.read().expect("Rumor store lock poisoned");
-        with_closure(list.get(key).and_then(|r| r.get(member_id)));
+        if let Some(sublist) = list.get(key) {
+            if let Some(rumor) = sublist.get(member_id) {
+                with_closure(rumor);
+            }
+        }
+    }
+
+    pub fn assert_rumor_is<P>(&self, key: &str, member_id: &str, mut predicate: P)
```

Comment:
That's because it's used in `butterfly/tests/rumor/election.rs`... Rust integration test files are each compiled as a separate crate, using the real crate as a dependency. As such, in integration tests, the real crate is compiled normally, and not for testing. If you add `#[cfg(test)]` to this function, it means the integration test doesn't have access to it.

Reply:
Ah, that makes sense. Do you think there's there any sort of pattern that could be applied to test-only helper functions in this case?

Reply:
I haven't come up with any yet... I ran into this same scenario refactoring some other stuff in Butterfly a while back. 

Ultimately, I think it's probably going to take some rethinking of how we're testing it 😦